src/util/compareThought.ts

```diff
@@ -126,14 +159,24 @@ export const compareFormatting = <T, U>(a: T, b: U): ComparatorValue => {
   return aIsHtml && !bIsHtml ? -1 : bIsHtml && !aIsHtml ? 1 : 0
 }
 
-/** A comparison function that sorts date strings. */
-export const compareDateStrings: ComparatorFunction<string> = (a: string, b: string) =>
-  compare(parseDate(a), parseDate(b))
+/** A comparison function that sorts date strings. Only handles date vs date comparisons. */
+export const compareDateStrings: ComparatorFunction<string> = (a: string, b: string) => {
+  const aIsDate = isDatePattern(a)
+  const bIsDate = isDatePattern(b)
```

Comment:
Great, I like how you've extended this function and followed the pattern of other comparison functions.